### DISCUSSION TITLE
Mermaid JS Support - Just work with references to the Mermaid Live URLs!

### DISCUSSION THREAD

**Discussion by pkcpkc** [community/community#34688] [Other Feature Feedback, Questions, & Ideas]
I really love your support for Mermaid UML diagrams within markdown files!

One suggestion to make both: not pollute markdown with Mermaid ML code and seamless integration with Mermaid editor.

Instead of including the Mermaid ML, e.g. via:
```mermaid
sequenceDiagram
    participant A as Architect
    participant M as Mermaid Live
    participant G as GitHub
    
    Note over A,G: Design diagram
    A->>M: Design diagram in WYSIWYG editor
    M->>M: Mermaid keeps state in URL with pako (pako.inflate(mermaidML))
    Note over A,G: Put diagram in GitHub readme.md
    A->>M: Copy Mermaid URL including pako state (https://mermaid.live/edit_pako:eN...)
    M->>A: Mermaid URL including pako state
    A->>G: Put Mermaid URL including pako state in GitHub readme.md
    Note over A,G: Reader reads and edits diagram
    G->>G: Render diagram internally via Mermaid engine and deflate via pako (pako.deflate(mermaidML))
    A->>G: User clicks on diagram
    G->>M: Open Mermeraid Live WYSIWYG editor in another frame with Mermaid URL including pako state
    A->>M: Directly adjust diagram
    Note over A,G: GOTO: "Put diagram in GitHub readme.md"
```

Alternate I:
Just integrate the Mermaid Live editor URL, that contains the diagrams contents as pako (https://github.com/nodeca/pako) zipped URL param and deflate it before rendering it on GitHub:
https://mermaid.live/edit#pako:eNqNU21r2zAQ_iuHPiWQOt_9oWDW4g3iZqQrpWAomnRJtNiSp5eMUPrfe7Kc5m0js8EY3XP3PM_d6Y0JI5HlzOHvgFrgneIry9taAz0dt14J1XHtoQDuoLBirTwKfxmvYrxC23IlYaa2eAkpI6RU_mv4mYLp-2A8gtmihWJS5nCHTq00yGMhxc3tbXUeAqXh-eXx2_NLCSiVNzaBqwTea9kgdg6c58RCGU-LGfxRfk3KNgZG8ZspvWwoPGpTSjUbj_8q7Xvwx-TJCljkssWslSdav5hu96khkiotmiCVXiXmJGi09r5z-XQ6UGcNdW4a3bxGVI4PWZaND76K_GrNg4pB8VUR_7RyZn9BUfqPIAdcy77r7nRSZeJdoI7QQ7M8Ws2bZgdbxT8VoV4pjX0pif0I-vDRZIbjy8kM_p4csYhGiY0Doy-l0CDmHeqeEe1-Nc_WJjaAa-PXVGtJ6ZgW5P8bHVdTWboW5I_LX8H5UyVnfSznP-Y51OzaOrFaxzdWYBM2tIDu6ls8qxnpbbFmsRK1iYfGx4x3goZOkr773hzLl7xxOGE8ePO404Ll3gbcg4b7PqDePwA_4Fky

Disadvantages:

- You don't see exact changes in Mermaid ML directly in the binary pako format, without opining it.
- Dependency with Mermaid Live

Alternate II:
At least allow to reference an external MermaidML .mmd file in the repository to keep MermaidML and Markdown separate.

Instead of:

```
```mermaid
```

Allow referencing the .mmd file:

```
'![My sequence diagram](./sequence.mmd)'
```

Alternate III:
Same as II, but via:

```
`![My sequence diagram](https://mermaid.live/edit#pako:eNqNU21r2zAQ_iuHPiWQOt_9oWDW4g3iZqQrpWAomnRJtNiSp5eMUPrfe7Kc5m0js8EY3XP3PM_d6Y0JI5HlzOHvgFrgneIry9taAz0dt14J1XHtoQDuoLBirTwKfxmvYrxC23IlYaa2eAkpI6RU_mv4mYLp-2A8gtmihWJS5nCHTq00yGMhxc3tbXUeAqXh-eXx2_NLCSiVNzaBqwTea9kgdg6c58RCGU-LGfxRfk3KNgZG8ZspvWwoPGpTSjUbj_8q7Xvwx-TJCljkssWslSdav5hu96khkiotmiCVXiXmJGi09r5z-XQ6UGcNdW4a3bxGVI4PWZaND76K_GrNg4pB8VUR_7RyZn9BUfqPIAdcy77r7nRSZeJdoI7QQ7M8Ws2bZgdbxT8VoV4pjX0pif0I-vDRZIbjy8kM_p4csYhGiY0Doy-l0CDmHeqeEe1-Nc_WJjaAa-PXVGtJ6ZgW5P8bHVdTWboW5I_LX8H5UyVnfSznP-Y51OzaOrFaxzdWYBM2tIDu6ls8qxnpbbFmsRK1iYfGx4x3goZOkr773hzLl7xxOGE8ePO404Ll3gbcg4b7PqDePwA_4Fky)`
```

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @pkcpkc ! We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Issues category is solely related to conversations around the GitHub products Issues and Projects. I've gone ahead and moved it for you. Good luck!
